rustls/src/client/client_conn.rs

```diff
@@ -969,15 +969,13 @@ impl std::error::Error for EarlyDataError {}
 #[derive(Debug)]
 pub struct ClientConnectionData {
     pub(super) early_data: EarlyData,
-    pub(super) resumption_ciphersuite: Option<SupportedCipherSuite>,
```

Comment:
I was interested in how/when this became dead code. ISTM that:

1. In 2021, with https://github.com/rustls/rustls/commit/b3d7417d3841a95dfd92635576d55ef62e7dda63, the field's primary use in `ClientConnection::negotiated_cipher_suite()` fell away. That left only one remaining read of the field, in the `impl quic::QuicExt for ClientConnection` trait implementation's `zero_rtt_keys()` fn.
2. In 2023, with https://github.com/rustls/rustls/commit/fc4e38ea22682cbd401ae30d03bc40c6bc69b406, the `zero_rtt_keys()` fn was changed to stop accessing that field when the code was migrated out of `rustls/src/client/client_conn.rs` and into `rustls/src/quic.rs`.